client/components/draft_input/draft_input_caption.coffee

```diff
@@ -0,0 +1,204 @@
+React = require 'react'
+window.global = window
+window.process = {env: {NODE_ENV: 'development'}}
+
+{ convertToRaw,
+  convertFromHTML,
+  CompositeDecorator,
+  ContentState,
+  Editor,
+  EditorState,
+  RichUtils,
+  Modifier,
+  getVisibleSelectionRect } = require 'draft-js'
+{ stateToHTML } = require 'draft-js-export-html'
+DraftDecorators = require './draft_decorators.coffee'
```

Comment:
Solid work wrangling the decorator pattern here! I'm assuming these are meant to be used as [ES7 decorators](https://medium.com/google-developers/exploring-es7-decorators-76ecb65fb841#.mgz2y2my5)? I find it unfortunate React devs are pushing this complicated JS feature so much, but glad to see you've tackled this.